pulse2percept/models/granley2021.py

```diff
@@ -867,3 +886,130 @@ def predict_percept(self, implant, t_percept=None):
+size_model: callable, optional
+    Model used to modulate percept size with amplitude, frequency, and
+    pulse duration
+streak_model: callable, optional
+    Model used to modulate percept streak length with amplitude, frequency,
+    and pulse duration
+do_thresholding: boolean
+    Use probabilistic sigmoid thresholding, default: False
+**params: dict, optional
+    Additional parameters to customize the model
+"""
+    def __init__(self, p2pmodel):
+        super().__init__(p2pmodel)
+        
+        self.rho = torch.tensor(p2pmodel.rho, device=self.device)
+        self.shape = p2pmodel.grid.shape
+        self.axon_contrib = torch.tensor(p2pmodel.axon_contrib, device=self.device)
+        self.rho = torch.tensor(p2pmodel.rho, device=self.device)
+        self.axlambda = torch.tensor(p2pmodel.axlambda, device=self.device)
+            
+        self.a0 = torch.tensor(p2pmodel.a0, device=self.device)
+        self.a1 = torch.tensor(p2pmodel.a1, device=self.device)
+        self.a2 = torch.tensor(p2pmodel.a2, device=self.device)
+        self.a3 = torch.tensor(p2pmodel.a3, device=self.device)
+        self.a4 = torch.tensor(p2pmodel.a4, device=self.device)
+        self.a5 = torch.tensor(p2pmodel.a5, device=self.device)
+        self.a6 = torch.tensor(p2pmodel.a6, device=self.device)
+        self.a7 = torch.tensor(p2pmodel.a7, device=self.device)
+        self.a8 = torch.tensor(p2pmodel.a8, device=self.device)
+        self.a9 = torch.tensor(p2pmodel.a9, device=self.device)
+        
+        self.locs = {}
+        # Other parameters from initialization args
+        self.percept_shape = p2pmodel.grid.shape  # p2pmodel has a grid attribute defining percept shape
+        self.thresh_percept = p2pmodel.thresh_percept
+    
+    
+    def forward(self, stims, e_locs, model_params=None):
+        if model_params is None:
+            rho = self.rho
+        else:
+            rho = model_params[0]
+        
+        if model_params is None:
+            axlambda = self.axlambda
+        else:
+            axlambda = model_params[1]
+
+        
+        a0 = self.a0 if model_params is None else model_params[2]
+        a1 = self.a1 if model_params is None else model_params[3]
+        a2 = self.a2 if model_params is None else model_params[4]
+        a3 = self.a3 if model_params is None else model_params[5]
+        a4 = self.a4 if model_params is None else model_params[6] # not needed
+        a5 = self.a5 if model_params is None else model_params[7]
+        a6 = self.a6 if model_params is None else model_params[8]
+        a7 = self.a7 if model_params is None else model_params[9]
+        a8 = self.a8 if model_params is None else model_params[10]
+        a9 = self.a9 if model_params is None else model_params[11]
+
+        # parse from stims
+        freq = stims[:,0] 
+        amp = stims[:,1]
+        pdur = stims[:,2]
+
+        scaled_amps = (a1 + a0*pdur) * amp
+
+        # bright
+        F_bright = a2 * scaled_amps + a3 * freq
```

Comment:
In my code i didnt use a4 but here you should. Should be `F_bright = a2 * scaled_amps + a3 * freq + a4`